testonly/map_metadata.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testonly
+
+import(
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+)
+
+// MustMarshalMetadata is used in tests to Marshal proto messages used as map
+// metadata into the protobuf.ptypes.any.Any used in the Trillian API and in
+// storage.  Failure to marshal will fail the test.
+func MustMarshalMetadata(t *testing.T, in proto.Message) *any.Any {
```

Comment:
Could you make this more general? Maybe `MustMarshalAny` / `testonly/any.go`?

Bonus points for replacing our [ever-growing army of mustMarshalAny functions](https://github.com/google/trillian/search?utf8=%E2%9C%93&q=mustMarshalAny&type=) (or adding a TODO).

Reply:
Oh I didn't realise how many times that's already been done! I'll go and fix that.

Reply:
Thanks!  :D